hours/importer/kirjastot.py

```
@@ -222,84 +292,363 @@ def get_openings(self, data: list) -> list:
+                    TimeElement(
+                        start_time=None,
+                        end_time=None,
+                        resource_state=State.CLOSED,
+                        override=True
+                        if schedule.get("period") in override_periods
+                        else False,
+                        full_day=True,
+                    )
+                )
+            else:
+                for schedule_time in schedule.get("times"):
+                    try:
+                        start_time = datetime.strptime(
+                            schedule_time.get("from"), "%H:%M"
+                        ).time()
+                    except ValueError:
+                        start_time = None
+                    try:
+                        end_time = datetime.strptime(
+                            schedule_time.get("to"), "%H:%M"
+                        ).time()
+                    except ValueError:
+                        end_time = None
+
+                    time_elements.append(
+                        TimeElement(
+                            start_time=start_time,
+                            end_time=end_time,
+                            resource_state=KIRKANTA_STATUS_MAP[schedule_time["status"]],
+                            override=True
+                            if schedule.get("period") in override_periods
+                            else False,
+                            full_day=False,
+                        )
+                    )
+
+            schedule_date = schedule.get("date")
+            if not isinstance(schedule_date, date):
+                schedule_date = parse(schedule.get("date")).date()
+
+            time_elements = combine_element_time_spans(time_elements)
+
+            time_elements.sort(key=self._get_times_for_sort)
+            opening_hours[schedule_date].sort(key=self._get_times_for_sort)
+
+            assert time_elements == opening_hours[schedule_date]
```

Probably we need to use something else than `assert` here, since production does not necessarily run with assertions turned on, and we probably always want to raise an exception in production if this happens.